// tome/index.js
// import React, { Component } from "react";
// export default class Graph extends Component {
const _ = require("lodash");

const articles = require("../articles/articles-compiled.json");

export default class Tome {
  constructor() {
    console.log("Hey from Tome!");

    // this.articles = this.parseArticles(articles);
    this.articles = articles;
  }
  // Parse the JSON for the HTML content (remove /n newline and stuff)
  // parseArticles(articles) {
  //   console.log(articles);
  //   return articles.map(_ => {
  //     _.content = JSON.parse(_.content);
  //     return _;
  //   });
  // }
  getHeader() {
    return "Testing node_module concept";
  }
  getArticles() {
    return this.articles;
  }

  // Satan's Revenge -> "satan%27s-revenge"
  titleToUrl(title) {
    // Satan's Revenge -> "Satan%27s%20Revenge"
    // return encodeURIComponent(str).replace(/[!'()*]/g, escape);

    // Satan's Revenge
    let url = title;
    //   // Satan's-Revenge
    //   .replace(/ /g, "-")
    //   // satan's-revenge
    //   .toLowerCase();

    // // satan's-revenge
    // url = encodeURIComponent(title);
    // console.log(url);

    // // satan%27s-revenge
    // url = title.replace(/[!'()*]/g, escape);

    url = _.kebabCase(url);

    return url;
  }

  // "satan%27s-revenge" -> Satan's Revenge
  urlToTitle(url) {}

  getArticleByTitle(title) {
    // return this.articles.filter(_ => _.title === title)[0];
    // Actually, be more vague about it
  }
}
