// tome/index.js
// import React, { Component } from "react";
// export default class Graph extends Component {
const _ = require("lodash");

const articles = require("../../src/articles/articles-compiled.json");

export default class Tome {
  constructor() {
    console.log("Hey from Tome!");

    // this.articles = this.parseArticles(articles);
    this.articles = articles;

    this.articles = this.createArticleLinks(this.articles);
  }

  getHeader() {
    return "Testing node_module concept";
  }

  getArticles() {
    return this.articles;
  }

  createArticleLinks(articles) {
    articles = articles.map(article => {
      article.url = "/" + this.titleToUrl(article.title);
      return article;
    });
    return articles;
  }

  // Satan's Revenge -> "satans-revenge"
  titleToUrl(title) {
    // Satan's Revenge -> "Satan%27s%20Revenge"
    // return encodeURIComponent(str).replace(/[!'()*]/g, escape);

    // Satan's Revenge
    let url = title;

    //   // Satan's-Revenge
    //   .replace(/ /g, "-")
    //   // satan's-revenge
    //   .toLowerCase();

    // // satan's-revenge
    // url = encodeURIComponent(title);
    // console.log(url);

    // // satan%27s-revenge
    // url = title.replace(/[!'()*]/g, escape);

    // satans-revenge
    url = _.kebabCase(url);

    return url;
  }

  getArticleByTitle(title) {
    // return this.articles.filter(_ => _.title === title)[0];
    // Actually, be more vague about it, matching the kebab cases
    return this.articles.filter(
      x => _.kebabCase(x.title) === _.kebabCase(title)
    )[0];
  }
}
